pymove/tests/preprocessing/test_filters.py

```diff
@@ -0,0 +1,567 @@
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[0, 2, 3]
+    )
+    assert_frame_equal(filter_out_values, expected)
+    assert move_df.len() == 4
+
+    filters.by_bbox(move_df, bbox, inplace=True)
+    expected = DataFrame(data=[
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1]
+    ], columns=cols, index=[1]
+    )
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 1
+
+
+def test_by_datetime():
+    move_df, cols = _prepare_df_default(list_data_1)
+
+    filter_values_start = filters.by_datetime(
+        move_df, start_datetime='2008-10-23 05:53:06')
+    expected = DataFrame(data=[
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[1, 2, 3]
+    )
+    assert_frame_equal(filter_values_start, expected)
+    assert move_df.len() == 4
+
+    filter_values_end = filters.by_datetime(move_df, end_datetime='2008-10-23 05:53:05')
+    expected = DataFrame(data=[
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1]
+    ], columns=cols, index=[0]
+    )
+    assert_frame_equal(filter_values_end, expected)
+    assert move_df.len() == 4
+
+    filter_values_start_end = filters.by_datetime(
+        move_df, start_datetime='2008-10-23 05:53:04', end_datetime='2008-10-23 05:53:06'
+    )
+    expected = DataFrame(data=[
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1]
+    ], columns=cols, index=[0, 1]
+    )
+    assert_frame_equal(filter_values_start_end, expected)
+    assert move_df.len() == 4
+
+    filter_out_values = filters.by_datetime(
+        move_df, start_datetime='2008-10-23 05:53:04', end_datetime='2008-10-23 05:53:06',
+        filter_out=True
+    )
+    expected = DataFrame([
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[2, 3]
+    )
+    assert_frame_equal(filter_out_values, expected)
+    assert move_df.len() == 4
+
+    filters.by_datetime(
+        move_df, start_datetime='2008-10-23 05:53:04', end_datetime='2008-10-23 05:53:06',
+        filter_out=True, inplace=True
+    )
+    expected = DataFrame([
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[2, 3]
+    )
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)